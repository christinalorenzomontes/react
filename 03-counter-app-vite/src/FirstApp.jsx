import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, name }) => {

    return (<>
        <h1>{ title }</h1>
        {/* <code>{ JSON.stringify( newMessage )}</code> */}
        {/* <p>{ sumF() }</p> */}
        <p>{ subtitle }</p>
        <p>{ name }</p>
    </>);

}
export default FirstApp;

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
}

FirstApp.defaultProps = {
    //title: 'No title',
    subtitle: 'No subtitle',
    name: 'Christina'
}