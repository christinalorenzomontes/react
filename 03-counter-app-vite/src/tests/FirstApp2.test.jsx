import { render, screen } from "@testing-library/react";
import FirstApp from "../FirstApp";

describe('Pruebas en <FirstApp2 />', () => {

    const title = 'Hola, soy Chris';
    const subtitle = 'Soy un subtítulo';
    
    test('Debe hacer match con el snapshot', () => {
        const { container } = render( <FirstApp title={ title } /> )
        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar el mensaje "Hola, soy Chris"', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('Debe mostrar el título en un h1', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
    });

    test('Debe mostrar el subtítulo enviado por props', () => {
        const { getAllByText } = render( 
            <FirstApp
                title={ title } 
                subtitle={ subtitle }    
            />
        );

        expect( getAllByText(subtitle).length ).toBe(2);
    });

});