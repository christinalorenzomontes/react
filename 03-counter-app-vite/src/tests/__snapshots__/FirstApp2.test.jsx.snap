// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp2 /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Chris
  </h1>
  <p>
    No subtitle
  </p>
  <p>
    No subtitle
  </p>
  <p>
    Christina
  </p>
</div>
`;
